### PR TITLE
zephyr: Add support for the nRF5340 Audio devkit

### PR DESCRIPTION
**Pull request by Frodevan**
Adds the nRF5340 audio devkit board, with capabilities to build both application and network cores with configurable separate builds.

### CODE REVIEW

File: autopts/ptsprojects/boards/nrf53_audio.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+                         debugger_snr,
+                         app_core_configs,
+                         True)
+
+    config_dir_net = os.getenv("AUTOPTS_SOURCE_DIR_NET")
+    if config_dir_net is None:
+        net_core_configs = [f'OVERLAY_CONFIG=\'nrf5340_cpunet_iso-bt_ll_sw_split.conf;'
+                            f'../../../tests/bluetooth/tester/nrf5340_hci_ipc_cpunet.conf\'']
+    else:
+        net_core_configs = [f'OVERLAY_CONFIG=\'../../../{config_dir_net}/hci_ipc.conf\'']
```
Comment (m-alperen-sener) [inline autopts/ptsprojects/boards/nrf53_audio.py:74]: Is it possible to use path as absolute rather than relative; or at least basing the path to zephyr_wd if it is possible?

**Review by m-alperen-sener — APPROVED** [review]
Seems reasonable; small comments.
Can we document AUTOPTS_SOURCE_DIR_NET and AUTOPTS_SOURCE_DIR_APP;

File: autopts/ptsprojects/boards/nrf53_audio.py (hunk 1 above)
Comment (Frodevan) [reply autopts/ptsprojects/boards/nrf53_audio.py:74]: I mean, AUTOPTS_RESET_CMD is not documented in the board_ext file. I am not sure a good place to put this documentation, since it only applies to one group of boards, or just one board in particular.

However, I can try to have the path absolute. The build here will in any ways always be referenced from the hci_ipc source folder, so /../../../ should always be the zephyr root.

Comment (Frodevan) [reply autopts/ptsprojects/boards/nrf53_audio.py]: The relative path should now have been changed to absolute

**Comment by Frodevan** [comment]
> Seems reasonable; small comments. Can we document AUTOPTS_SOURCE_DIR_NET and AUTOPTS_SOURCE_DIR_APP;

I mean, AUTOPTS_RESET_CMD is not documented in the board_ext file. I am not sure a good place to put this documentation, since it only applies to one group of boards, or just one board in particular.

**Review by theob-pro — APPROVED** [review]
👍

File: autopts/ptsprojects/boards/__init__.py [hunk 2]
```diff
@@ -204,8 +211,8 @@ def get_tty(debugger_snr):
     for dev in devices.keys():
         if devices[dev] == debugger_snr:
             tty = dev
-            break
```
Comment (sjanc) [inline autopts/ptsprojects/boards/__init__.py:207]: why keep iterating here?

File: autopts/ptsprojects/zephyr/iutctl.py [hunk 3]
```diff
@@ -205,6 +205,7 @@ def rtt_logger_start(self):
 
     def rtt_logger_stop(self):
         if self.rtt_logger:
+            time.sleep(0.1)
```
Comment (sjanc) [inline autopts/ptsprojects/zephyr/iutctl.py:208]: why is this needed?

**Review by sjanc — COMMENTED** [review]
I like the changes in general code, and we should also extend nrf5340DK same way,

but code changes are not matching commit message,  since there are more features than just audioDK support,  would be good to maybe split it for commit with proper commit message so that it is easier to figure eg. why changes to get_free_device() or get_tty() are needed

File: autopts/ptsprojects/zephyr/iutctl.py (hunk 3 above)
Comment (Frodevan) [reply autopts/ptsprojects/zephyr/iutctl.py:208]: In the case where an issue in the host has caused an operation to fail and in turn a test-case to fail, the rtt logger will often not have the time to pick up an error-print from the IUT before being closed. When this happens, the IUT logs will be incomplete, since any hints to the actual reason of the failure will be missing. This delay is to ensure enough time to prevent such a situation.

File: autopts/ptsprojects/boards/__init__.py (hunk 2 above)
Comment (Frodevan) [reply autopts/ptsprojects/boards/__init__.py:207]: This may not be needed, but at the time I was adding that change I seem to remember that there was problems where it was picking the UART for the wrong core of the nRF5340 chip.

File: autopts/ptsprojects/zephyr/iutctl.py (hunk 3 above)
Comment (sjanc) [reply autopts/ptsprojects/zephyr/iutctl.py:208]: OK, I guess short comment in code would be good to not forget about this :)

Comment (Frodevan) [reply autopts/ptsprojects/zephyr/iutctl.py]: Comment added

File: autopts/ptsprojects/boards/__init__.py (hunk 2 above)
Comment (Frodevan) [reply autopts/ptsprojects/boards/__init__.py:207]: I added the conditional check, with a comment which should spread some light on why the change is needed to support the nRF5340 Audio devkit.

**Review by sjanc — APPROVED** [review]

File: autopts/ptsprojects/zephyr/iutctl.py (hunk 3 above)
Comment (sjanc) [reply autopts/ptsprojects/zephyr/iutctl.py]: btw with assert message it seems that 0.1 is not enough
